Improved Login Page UI

onlyVishesh opened the pull request:
I have added Code of the given login page Figma UI and also i includes signup option in it.
Please review my PR
Thank You

ShreyaPrasad1209 reviewed (changes requested):
Another thing to keep in mind - whenever we add a sub-component inside a component we should create a subfolder for easy code maintenance. Eg: Social logins could be moved inside another folder inside Footer. 

- Footer
      - SocialMediaIcons

This is just an example, we don't actually need a separate component for SocialMediaIcons since they'll always remain a part of Footer.

onlyVishesh commented on src/components/Footer/main.tsx:
```diff
@@ -1,65 +1,94 @@
+            respective owners.
+          </TitleP>
+        </Title>
+        <Socials>
```
I used svg directly because a made this page using figma frame and it is easy to directly use svg but i will change svg with @fortawesome/free-brands-svg-icons in my next commit.

onlyVishesh commented on src/components/LoginPage/styled.tsx:
```diff
@@ -1,20 +1,64 @@
-  margin: 5px;
-  border: 1px solid #ccc;
-  border-radius: 5px;
+export const Left = styled.div`
```
@ShreyaPrasad1209  Ma'am should i remove all the unsued files like left.tsx, rightHeading,tsx and their style as we are not using them or should i leave them for future use ?

onlyVishesh replied:
I have changed the UI of login page as you told me

ShreyaPrasad1209 replied:
> @ShreyaPrasad1209 Ma'am should i remove all the unsued files like Left.tsx, RightHeading,tsx and their style as we are not using them or should i leave them for future use ?

@onlyVishesh remove them

onlyVishesh commented:
@ShreyaPrasad1209 Maam i have updated the code with all your suggestions, UI changes and all the other changes you have asked like - 
1. Changed UI of login Page
2. Aligned Navbar items and enhance the login button's UI ( i made small change in the ui of login button when we hover over it)
3. Break down Footer to individual components and use @fortawesome/free-brands-svg-icons in place of svg.

Is there anything else you want to add or remove ? 
( Ma'am i haven't remove the unsued files like Left.tsx, RightHeading,tsx and their style. So should i leave them for future use or remove them from the code ? )

ShreyaPrasad1209 commented on src/components/Footer/SocialStyled.tsx:
```diff
@@ -0,0 +1,20 @@
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  gap: 25px;
```
can we use rem instead of px?

ShreyaPrasad1209 commented on src/components/Footer/SocialStyled.tsx:
```diff
@@ -0,0 +1,20 @@
+  cursor: pointer;
+
+  @media only screen and (min-width: 1025px) and (max-width: 1280px) {
+    width: 40px;
```
avoid using px, please use relative units.

ShreyaPrasad1209 commented on src/components/Footer/main.tsx:
```diff
@@ -1,65 +1,94 @@
+        <Socials>
+          <a href="/">
+            <Svg
+              xmlns="http://www.w3.org/2000/svg"
```
do we need these svgs anymore now that we're using social icons?

ShreyaPrasad1209 commented on src/components/Footer/styled.tsx:
```diff
@@ -1,66 +1,116 @@
+  background: #000;
   position: absolute;
   bottom: 0;
+  padding: 0% 20%;
```
why is padding needed?

ShreyaPrasad1209 commented on src/components/Footer/styled.tsx:
```diff
@@ -1,66 +1,116 @@
+  @media only screen and (max-width: 600px) {
+    width: 100%;
+    flex-direction: column;
+    gap: 2px;
```
rem/percentage not px. Please look up on good practices for gap values.

ShreyaPrasad1209 commented on src/components/LoginPage/LeftSide.tsx:
you can remove this, no longer needed

ShreyaPrasad1209 commented on src/components/LoginPage/LoginForm.tsx:
```diff
@@ -0,0 +1,77 @@
+  };
+
+  return (
+    <>
```
It's a good practice to use div over empty JSX fragments due to accessibility reasons.

ShreyaPrasad1209 commented on src/components/LoginPage/LoginFormStyled.tsx:
```diff
@@ -0,0 +1,159 @@
+import styled from 'styled-components';
+
+export const TabBtn = styled.div`
+  margin: 15px 0 10px 0;
```
why is margin required?

ShreyaPrasad1209 commented on src/components/LoginPage/LoginFormStyled.tsx:
```diff
@@ -0,0 +1,159 @@
+import styled from 'styled-components';
+
+export const TabBtn = styled.div`
```
A descriptive name could be ToggleButton, what do you think?

ShreyaPrasad1209 commented on src/components/LoginPage/LoginFormStyled.tsx:
```diff
@@ -0,0 +1,159 @@
+  }
+`;
+
+export const LoginButton = styled.button`
```
separate the styling for a normal button and then reuse it for Login using styled components. Here's how you can reuse it: https://styled-components.com/docs/basics#extending-styles

ShreyaPrasad1209 commented on src/components/LoginPage/LoginFormStyled.tsx:
```diff
@@ -0,0 +1,159 @@
+  cursor: pointer;
+`;
+
+export const Forgot = styled.p`
```
by separating a generic button's CSS and extending it to different buttons, we are reusing the same code and writing less, maintainable code. Like how we can simply extend the base Button's CSS and apply custom styling here. Same goes for other buttons in the code.

ShreyaPrasad1209 commented on src/components/LoginPage/Right.tsx:
no longer needed

ShreyaPrasad1209 commented on src/components/LoginPage/RightHeading.tsx:
no longer needed

ShreyaPrasad1209 commented on src/components/LoginPage/RightStyled.tsx:
no longer needed

ShreyaPrasad1209 commented on src/components/LoginPage/main.tsx:
```diff
@@ -1,29 +1,19 @@
-    handleLogin(username, password);
-  };
+import Right from './Right';
+// import LeftSide from './LeftSide';
```
please remove commented code

ShreyaPrasad1209 reviewed (changes requested):
I've reviewed as much as possible, please make these changes first and then I'll move on to next set of files as it's a huge PR.
